kubernetes/resource_kubernetes_default_service_account.go

```diff
@@ -59,6 +60,25 @@ func resourceKubernetesDefaultServiceAccountCreate(ctx context.Context, d *schem
 		return diag.FromErr(err)
 	}
 
+	if v, ok := d.GetOkExists("automount_service_account_token"); ok {
+		ops := patchMetadata("metadata.0.", "/metadata/", d)
+		ops = append(ops, &ReplaceOperation{
+			Path:  "/automountServiceAccountToken",
+			Value: v.(bool),
+		})
+
+		data, err := ops.MarshalJSON()
+		if err != nil {
+			return diag.Errorf("Failed to marshal update operations: %s", err)
+		}
+		log.Printf("[INFO] Updating default service account %q: %v", metadata.Name, string(data))
+		out, err := conn.CoreV1().ServiceAccounts(metadata.Namespace).Patch(ctx, metadata.Name, pkgApi.JSONPatchType, data, metav1.PatchOptions{})
+		if err != nil {
+			return diag.Errorf("Failed to update default service account: %s", err)
+		}
+		log.Printf("[INFO] Submitted updated default service account: %#v", out)
+	}
+
```

Comment:
With the above changes in place, we don't need to call the update function at the end of the create function. Instead, we can go ahead and call the read function right away to get the current state of the resource from Kubernetes.

Could you please replace `resourceKubernetesServiceAccountUpdate` with `resourceKubernetesServiceAccountRead`?

Reply:
I thought update function is still needed for the other arguments like `image_pull_secret` or `secret` but as you suggested I'll move them to this resource and call `resourceKubernetesServiceAccountRead`.